src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/BookingExtractionService.kt

```diff
@@ -346,23 +378,30 @@ class BookingExtractionService(
         }
       }
     }
+  }
 
+  private fun extractErsedForBooking(
+    sentences: List<CalculableSentence>,
+    breakdownByReleaseDateType: MutableMap<ReleaseDateType, ReleaseDateCalculationBreakdown>,
+    dates: MutableMap<ReleaseDateType, LocalDate>,
+    latestReleaseDate: LocalDate,
+  ) {
     val latestEarlyReleaseSchemeEligibilitySentence =
       extractionService.mostRecentSentenceOrNull(sentences, SentenceCalculation::earlyReleaseSchemeEligibilityDate) { !it.sentenceCalculation.isImmediateRelease() }
     val latestAFineRelease =
-      extractionService.mostRecentOrNull(sentences.filterIsInstance<AFineSentence>(), SentenceCalculation::releaseDate)
-    val afineIsRelease = latestAFineRelease == latestReleaseDate
+      extractionService.mostRecentSentenceOrNull(sentences.filterIsInstance<AFineSentence>(), SentenceCalculation::releaseDate)
+    val afineIsRelease = latestAFineRelease?.sentenceCalculation?.releaseDate == latestReleaseDate
     if (latestEarlyReleaseSchemeEligibilitySentence != null && !afineIsRelease) {
       if (latestAFineRelease != null && latestEarlyReleaseSchemeEligibilitySentence.sentenceCalculation.earlyReleaseSchemeEligibilityDate!!.isBefore(
-          latestAFineRelease,
-        )
+          latestAFineRelease.sentenceCalculation.releaseDate,
+        ) && !latestAFineRelease.sentenceCalculation.isImmediateRelease()
```

Comment:
just checking as I don't understand the logic. If there is another afine that is not an immediate release should we adjust to that?

Reply:
Yes as there is time to be served on it still and it's a term of imprisonment 
It's a bit odd how it's treated differently in some scenarios but not in others

Reply:
I plan on adding some tests fir the extracted methods today to make it clearer

Reply:
Just thought I can get as in in now with existing coverage of checking the overall output